materials/homework/list/index.html

```diff
@@ -0,0 +1,213 @@
+
+        Postup:
+        <ul>
+            <li>Máš seznam <strong>hodnot</strong>, které chceš seřadit podle
+                nějakého <strong>klíče</strong>. Klíč se dá z každé hodnoty
+                vypočítat.
+            </li>
+            <li>Vytvoř seznam dvojic <code>(klíč, hodnota)</code>.</li>
+            <li>
+                Seřaď tento seznam dvojic – dvojice se řadí nejdřív podle
+                prvního prvku, pak druhého atd.
+            </li>
+            <li>Nakonec vytvoř ze seznamu dvojic opět jen seznam hodnot.</li>
+        </ul>
+        Proč má zrovna had takovéhle výsadní postavení, zjistíš později.
+    </li>
+</ol>
+
+
+<h2>Římské číslice</h2>
+Jedna klasická programovací úloha, která nejspíš pořádně potrápí tvé logické
+myšlení. Je nepovinná, nemáš-li na ni aspoň pár hodin, tak ji přeskoč.
+
+<ol start="7">
+    <li>
+        Napiš funkci, která převede římské číslice na číslo (<code>int</code>).
+        <br>
+        <em>Tip: Nejdřív napiš k této funkci nějaké testy, aby sis ověřila, že
+        (a co) (ne)funguje správně.</em>
+    </li>
+</ol>
+
+
+<h2>Dadaistický koutek</h2>
+Procvičení práce se seznamy, řetězci a soubory. Opět: nebudeš-li vědět jak dál,
+zeptej se!
+
+<ol start="8">
+    <li>
+        Vyber si básničku, která má aspoň tři sloky po aspoň třech verších.
+        Ulož ji do souboru <code>basnicka.txt</code>.
+    </li>
+    <li>
+        Napiš program, který vypíše básničku ze souboru
+        <code>basnicka.txt</code>, ale obrátí pořadí veršů (t.j. jako první
+        vypíše poslední řádek, atd.)<br>
+        <em>Nápověda: Každý seznam má metodu <code>reverse</code>, která ho
+        „obrátí”.</em>
+    </li>
+    <li>Napiš program, který obrátí pořadí slov v jednotlivých verších.</li>
+    <li>
+        Obrať pořadí slok (ty by měly být oddělené jedním prázdným řádkem).
+    </li>
+    <li>
+        Vypiš slova básně v náhodném pořadí.<br>
+        <em>Bonusový projekt: Snaž se přitom co nejlépe zachovat strukturu
+        básně (sloky, verše, interpunkci, velká písmena, ...).</em>
+    </li>
+</ol>
+
+
+<h2>Had</h2>
+Následující projekty závisí jeden na druhém, řeš je postupně. Až to uděláš,
+můžeš si zahrát hru! Tahle sekce není jednoduchá (a poslední dva projekty jsou
+obzvláště náročné). Kdyžtak můžeš zkusit spojit síly s ostatními účastnicemi
+kurzu. :)
+
+<ol start="13">
+    <li>
+        Napiš funkci, která dostane seznam souřadnic (párů čísel menších než
+        10) a vypíše je jako mapu. Například:
+<pre>
+nakresli_mapu([(0, 0), (1, 0), (2, 2), (4, 3), (8, 9)])
+X X . . . . . . . .
+. . . . . . . . . .
+. . X . . . . . . .
+. . . . X . . . . .
+. . . . . . . . . .
+. . . . . . . . . .
+. . . . . . . . . .
+. . . . . . . . . .
+. . . . . . . . . .
+. . . . . . . . X .
+</pre>
+        <em>Jak na to?
+        <ul>
```

Comment:
This is not valid HTML: a block-level element (`<ul>`) cannot be inside an inline element (`<em>`).